BEST_PRACTICES.md

```diff
@@ -22,29 +41,23 @@ The recommended version of Xcode (and Swift) for all purposes is the current ava
 
 Enable warnings by adding `-Weverything` to your Build Settings under "Other Compiler Flags". If you need to ignore a specific warning you can use [Clang's pragma feature](http://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-via-pragmas) or add `-Wno-warning-to-be-disabled` (for example `-Wno-gnu-conditional-omitted-operand`).
 
-## Deployment
+## Versioning
 
-### Semantic Versioning
-
-We support [semantic versioning](http://semver.org/), and it's important that minor releases are backwards compatible otherwise don't feel shy to make it a major release.
+To version our open source projects we use [semantic versioning](http://semver.org/), and it's important that minor releases are backwards compatible otherwise don't feel shy to make it a major release.
 
 When making backwards compatible changes, flag your old APIs as deprecated like this:
 
-**Objective-C**:
-```objc
-- (NSInteger)objectWithIndex:(NSInteger)bar __attribute__((deprecated("Use objectAtIndex: instead")));
-```
-
-**Swift**:
 ```swift
 @available(*, deprecated=4.3.0, message="Use `objectAt(index index: Int)` instead") public func objectAtIndex(index: Int)
 ```
 
+When it comes to apps, patch releases are bug fixes, minor releases are small new features and major releases are re-designs or big features.
+
 ## Comments
 
 When they are needed, comments should be used to explain **why** a particular piece of code does something instead of **what**. Any comments that are used must be kept up-to-date or deleted.
 
-Block comments should generally be avoided, as code should be as self-documenting as possible, with only the need for intermittent, few-line explanations. This does not apply to those comments used to generate documentation.
+Comments should generally be avoided, since they tend to be abused and most developers use it to state the obvious. This does not apply to those comments used to generate documentation.
```

Comment:
What about this:

**Preferred:**

```swift
/*
  Workaround: So far this is the scroll-to-bottom method that worked best,
  with proper animation, and no issues so far, regardless of scrollview content size.
  If this looks and feels like a hack, it's because it kinda is.
  By creating a rect that's 1x1, pointed at the bottom-right side of the scrollview's content
  and telling it to scroll there, regardless of insets and offsets, it will scroll to the very bottom.
*/
func scrollToBottom() {
    let contentSize = self.collectionView.contentSize
    let bottomRect = CGRect(x: contentSize.width - 1, y: contentSize.height - 1, width: 1, height: 1)
    let visibleRect = self.collectionView.layer.visibleRect
    if !visibleRect.intersects(bottomRect) {
        self.collectionView.scrollRectToVisible(bottomRect, animated: true)
    }
}
```

**Not Preferred:**

```swift
// Scrolls to the bottom of the view
func scrollToBottom() {
    let contentSize = self.collectionView.contentSize
    let bottomRect = CGRect(x: contentSize.width - 1, y: contentSize.height - 1, width: 1, height: 1)
    let visibleRect = self.collectionView.layer.visibleRect
    if !visibleRect.intersects(bottomRect) {
        self.collectionView.scrollRectToVisible(bottomRect, animated: true)
    }
}
```